### PR TITLE
Add isGarbageCollected to perfscale-load-generator ObjListWatcher

### LINKED ISSUE #8025: Handle VMI cleanup in the load-generator tool
The load-generator tool creates many VMIs, waits until they are running, deletes the vmis, and waits until they are no longer in Running phase.  Since we're not guaranteeing the VMI objects are removed, sometimes VMIs hang around too long and the `make cleanup` recipe catches them in CI.   We shouldn't be relying on `make cleanup` because the load-generator should be able to cleanup after itself.  This will be very important when we want to do steady-state testing, where we rely heavily on a precise VMI count.

Example error: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/periodic-kubevirt-performance-cluster-scale-density-test/1542171219049058304

https://github.com/kubevirt/kubevirt/tree/main/tools/perfscale-load-generator

### PR DESCRIPTION
**Pull request by alaypatel07**
**What this PR does / why we need it**:

This helps in asserting the Delete semantics of the api during
performance testing. It is required that when Delete request is processed
against kubevirt api objects (VMI, VM and VirtualMachineReplicaSet),
kubevirt system is able to observe the complete garbage collection
(from delete request all the way to finalizer being removed and object being
removed from the kube api)

In order to avoid stomping apiserver with get requests from the e2e test suite
a further optimization of introducing a cache is required

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8025 

**Special notes for your reviewer**:

**Release note**:

```release-note
Wait deletion to succeed all the way till objects are finalized in perfscale tests
```

### CODE REVIEW

**Comment by alaypatel07** [comment]
/test pull-kubevirt-build

```
selecting docker as container runtime
dial tcp [::1]:8080: connect: connection refused
Build is not clean:
selecting docker as container runtime
Client Version: version.Info{GitVersion:"v0.55.0-124+df43661c4bc810-dirty", GitCommit:"df43661c4bc81015dd920e5747cce2d2c1a24ca6", GitTreeState:"dirty", BuildDate:"2022-07-21T17:34:41Z", GoVersion:"go1.17.8", Compiler:"gc", Platform:"linux/amd64"}
dial tcp [::1]:8080: connect: connection refused
dial tcp [::1]:8080: connect: connection refused
On branch main
Changes not staged for commit:
```

Docker did not come up for some reason. trying to see if this is a flake..

**Comment by alaypatel07** [comment]
https://asciinema.org/a/MNURPsH2alH3EzQ6JIJv6eJBt

I was able to make a local run wait for all the deletions with this.

**Comment by alaypatel07** [comment]
/test periodic-kubevirt-performance-cluster-100-density-test

trying to see if I can trigger the performance test on this PR

**Review by rthallisey — COMMENTED** [review]
One nit

File: tools/perfscale-load-generator/watcher/watcher.go [hunk 1]
```diff
@@ -122,10 +119,10 @@ func (w *ObjListWatcher) handleUpdate(obj interface{}) {
 			log.Log.V(2).Errorf("Could not convert VirtualMachineInstance obj: %v", w.ResourceKind)
 		}
 		if vmi.Status.Phase == kvv1.Running {
-			w.addObj(string(vmi.GetUID()))
+			w.addObj(w.getID(vmi))
 		}
-		if vmi.Status.Phase == kvv1.Succeeded {
-			w.deleteObj(string(vmi.GetUID()))
+		if vmi.Status.Phase == kvv1.Succeeded || vmi.Status.Phase == kvv1.Failed {
+			log.Log.V(4).Errorf("VirtualMachineInstance obj: %v Succeeded, waiting for garbage collection", fmt.Sprintf("%s/%s", vmi.Namespace, vmi.Name))
```
[the remark this answers is not in the document]
Comment (alaypatel07) [reply tools/perfscale-load-generator/watcher/watcher.go]: Fixed, thanks for the catch @rthallisey

**Comment by alaypatel07** [comment]
```
19:13:21:   Unexpected error:
19:13:21:       <*url.Error | 0xc001e415c0>: {
19:13:21:           Op: "Get",
19:13:21:           URL: "https://api.github.com/repos/kubevirt/kubevirt/releases?per_page=10000",
19:13:21:           Err: <http.tlsHandshakeTimeoutError>{},
19:13:21:       }
19:13:21:       Get "https://api.github.com/repos/kubevirt/kubevirt/releases?per_page=10000": net/http: TLS handshake timeout
19:13:21:   occurred
```
seems like a flak in getting releases https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/8027/pull-kubevirt-e2e-k8s-1.23-operator/1552675673892458496 

/test pull-kubevirt-e2e-k8s-1.23-operator

**Comment by alaypatel07** [comment]
reported the failure for this test here https://github.com/kubevirt/kubevirt/pull/7802#issuecomment-1198610148, retrying...

/test pull-kubevirt-e2e-kind-1.23-vgpu-nonroot

**Comment by alaypatel07** [comment]
reported here the failure here: https://github.com/kubevirt/kubevirt/pull/7168#issuecomment-1198636039

/test pull-kubevirt-e2e-k8s-1.23-sig-compute-migrations-nonroot

**Comment by alaypatel07** [comment]
/override pull-kubevirt-fossa

this failure is well-known and being discussed in the mailing list

**Pull request merged**